app/javascript/dashboard/routes/dashboard/settings/macros/Index.vue

```diff
@@ -1,117 +1,112 @@
-          <p class="flex flex-col items-center justify-center h-full">
-            {{ $t('MACROS.LIST.404') }}
-          </p>
-        </div>
-        <woot-loading-state
-          v-if="uiFlags.isFetching"
-          :message="$t('MACROS.LOADING')"
-        />
-        <table v-if="!uiFlags.isFetching && records.length" class="woot-table">
-          <thead>
-            <th
-              v-for="thHeader in $t('MACROS.LIST.TABLE_HEADER')"
-              :key="thHeader"
-            >
-              {{ thHeader }}
-            </th>
-          </thead>
-          <tbody>
-            <MacrosTableRow
-              v-for="(macro, index) in records"
-              :key="index"
-              :macro="macro"
-              @delete="openDeletePopup(macro, index)"
-            />
-          </tbody>
-        </table>
-      </div>
-      <div class="hidden w-1/3 lg:block">
-        <span v-dompurify-html="$t('MACROS.SIDEBAR_TXT')" />
-      </div>
-    </div>
-    <woot-delete-modal
-      :show.sync="showDeleteConfirmationPopup"
-      :on-close="closeDeletePopup"
-      :on-confirm="confirmDeletion"
-      :title="$t('LABEL_MGMT.DELETE.CONFIRM.TITLE')"
-      :message="$t('MACROS.DELETE.CONFIRM.MESSAGE')"
-      :message-value="deleteMessage"
-      :confirm-text="$t('MACROS.DELETE.CONFIRM.YES')"
-      :reject-text="$t('MACROS.DELETE.CONFIRM.NO')"
-    />
-  </div>
+  <SettingsLayout
+    :no-records-message="$t('MACROS.LIST.404')"
+    :no-records-found="!records.length"
+    :is-loading="uiFlags.isFetching"
+    :loading-message="$t('MACROS.LOADING')"
+    feature-name="macros"
+  >
+    <template #header>
+      <BaseSettingsHeader
+        :title="$t('MACROS.HEADER')"
+        :description="$t('MACROS.DESCRIPTION')"
+        :link-text="$t('MACROS.LEARN_MORE')"
+        feature-name="macros"
+      >
+        <template #actions>
+          <router-link
+            :to="{ name: 'macros_new' }"
+            class="button rounded-md primary"
+          >
+            <fluent-icon icon="add-circle" />
+            <span class="button__content">
+              {{ $t('MACROS.HEADER_BTN_TXT') }}
+            </span>
+          </router-link>
+        </template>
+      </BaseSettingsHeader>
+    </template>
+    <template #body>
+      <table class="min-w-full divide-y divide-slate-75 dark:divide-slate-700">
+        <thead>
+          <th
+            v-for="thHeader in $t('MACROS.LIST.TABLE_HEADER')"
+            :key="thHeader"
+            class="py-4 pr-4 text-left font-semibold text-slate-700 dark:text-slate-300"
+          >
+            {{ thHeader }}
+          </th>
+        </thead>
+        <tbody
+          class="divide-y divide-slate-50 dark:divide-slate-800 text-slate-700 dark:text-slate-300"
+        >
+          <MacrosTableRow
+            v-for="(macro, index) in records"
+            :key="index"
+            :macro="macro"
+            @delete="openDeletePopup(macro, index)"
```

Comment:
```suggestion
            @delete="openDeletePopup(macro)"
```

Index param is not used.

Reply:
Done.